tests/unit_tests/enrichment_test.py

```diff
@@ -16,6 +16,24 @@ def test_entire_enrichment_analysis(generate_proteins):
     assert isinstance(result, pd.DataFrame)
 
 
+def test_subset_enrichment_analysis(generate_proteins):
+    """Check that the test enrichment function returns a dataframe.
+    This test will run the GO enrichment on a subset of terms."""
+    df = generate_proteins
+    df.set_index("Protein", inplace=True)
+    terms = [
+        "nucleus",
+        "nucleoplasm",
+        "euchromatin",
+        "heterochromatin",
+        "protein-DNA complex",
+        "lysosome",
+        "cytoplasm",
+    ]
+    result = enrichment.test_enrichment(df, go_subset=terms)
+    assert isinstance(result, pd.DataFrame)
```

Comment:
**todo:** Check that only the terms you wanted to test are here.

Reply:
Good catch. That is fixed.